packages/cli/src/cmds/archive/analyze.ts

```diff
@@ -1,42 +1,119 @@
+
+  const oversizedAppMaps = new Set<string>();
+
+  // This could be done in worker threads as well, but the timing data indicates that it's not currently a bottleneck.
   {
-    const startTime = new Date().getTime();
     console.log(`Indexing AppMaps...`);
+    const startTime = new Date().getTime();
     const numIndexed = await indexAppMaps(appMapDir, maxAppMapSizeInBytes);
     const elapsed = new Date().getTime() - startTime;
     console.log(`Indexed ${numIndexed} AppMaps in ${elapsed}ms`);
   }
-  {
-    console.log('Generating sequence diagrams...');
-    const result = await updateSequenceDiagrams(appMapDir, maxAppMapSizeInBytes, compareFilter);
-    process.stdout.write(`done (${result.numGenerated})\n`);
-    oversizedAppMaps = result.oversizedAppMaps;
-  }
+
   {
     const startTime = new Date().getTime();
     console.log('Generating OpenAPI...');
     await generateOpenAPI(appMapDir, maxAppMapSizeInBytes);
     const elapsed = new Date().getTime() - startTime;
     console.log(`Generated OpenAPI in ${elapsed}ms`);
   }
-  {
+
+  const generateSequenceDiagrams = async () => {
+    const specOptions = {
+      loops: true,
+    } as SequenceDiagramOptions;
+
+    const task = (file: string): SequenceDiagramTask => ({
+      name: 'sequence-diagram',
+      verbose: verbose(),
+      appmapFile: file,
+      maxSize: maxAppMapSizeInBytes,
+      compareFilter,
+      specOptions,
+    });
+
+    console.log('Generating sequence diagrams...');
+    const startTime = new Date().getTime();
+
+    const result = await processAppMapDir<SequenceDiagramTask, SequenceDiagramResult>(
+      'Generate sequence diagrams',
+      workerPool,
+      task,
+      appMapDir
+    );
+
+    const elapsed = new Date().getTime() - startTime;
+    console.log(`Generated ${result.numProcessed} sequence diagrams in ${elapsed}ms`);
+    for (const file of result.oversized) oversizedAppMaps.add(file);
+  };
+
+  const scan = async () => {
+    let findingsCount = 0;
     const startTime = new Date().getTime();
     console.log('Scanning...');
-    const result = await scan(appMapDir);
+
+    const task = (file: string): ScanTask => ({
+      name: 'scan',
+      verbose: verbose(),
+      appmapFile: file,
+      maxSize: maxAppMapSizeInBytes,
+    });
+
+    const result = await processAppMapDir<ScanTask, ScanResult>(
+      'Scanning',
+      workerPool,
+      task,
+      appMapDir,
+      (_appmapFile: string, result: ScanResult) => {
+        if (result.findingsCount) findingsCount += result.findingsCount;
+      }
+    );
+
     const elapsed = new Date().getTime() - startTime;
-    console.log(`Scanned ${result} AppMaps in ${elapsed}ms`);
+    console.log(
+      `Scanned ${result.numProcessed} AppMaps in ${elapsed}ms, found ${findingsCount} findings`
+    );
+    for (const file of result.oversized) oversizedAppMaps.add(file);
+  };
+
+  assert(WORKER_FILE);
+  const workerPool = new WorkerPool(WORKER_FILE, threadCount);
```

Comment:
By using one worker pool for both sequence diagrams and scanning, we save a bunch of time that would be consumed by starting the worker threads twice. It takes somewhere between 0.5 and 2 seconds to start a worker thread and process a job.